CCIP: USDC guide (#1720)
smartcontractkit/documentation: src/content/ccip/tutorials/usdc.mdx

---
section: ccip
date: Last Modified
title: "Transfer USDC and Data Across Chains"
---

import { Aside, ClickToZoom, CodeSample, CopyText } from "@components"

USDC is a digital dollar backed 100% and is always redeemable 1:1 for US dollars. The [stablecoin](https://chain.link/education-hub/stablecoins) is issued by [Circle](https://www.circle.com/en/usdc) on multiple blockchain platforms.

This guide will first explain how Chainlink CCIP enables native USDC transfers under the hood by leveraging [Circle's Cross-Chain Transfer Protocol (CCTP)](https://www.circle.com/en/cross-chain-transfer-protocol). Then, you will learn how to use Chainlink CCIP to transfer USDC and arbitrary data from a smart contract on _Avalanche Fuji_ to a smart contract on _Polygon Mumbai_.

> **Note**: You are not limited to programmable token transfers but can use CCIP to transfer USDC tokens without data. Check the [Mainnets](/ccip/supported-networks/mainnet) and [Testnets](/ccip/supported-networks/testnet) configuration pages to learn on which blockchains CCIP supports USDC transfers.

Reviewer: ```suggestion
**Note**: In addition to programmable token transfers, you can also use CCIP to transfer USDC tokens without data. Check the [Mainnets](/ccip/supported-networks/mainnet) and [Testnets](/ccip/supported-networks/testnet) configuration pages to learn on which blockchains CCIP supports USDC transfers.
```

Reviewer: I also considered adding that programmable token transfers = tokens + data, since that association might not be immediately apparent to newer CCIP users. But if this guide is meant for more advanced users who would already know that, disregard

### Before you begin

1. You should understand how to write, compile, deploy, and fund a smart contract. If you need to brush up on the basics, read this [tutorial](/quickstarts/deploy-your-first-contract), which will guide you through using the [Solidity programming language](https://soliditylang.org/), interacting with the [MetaMask wallet](https://metamask.io) and working within the [Remix Development Environment](https://remix.ethereum.org/).
1. Your account must have some AVAX and LINK tokens on _Avalanche Fuji_ and MATIC tokens on _Polygon Mumbai_. You can use the [Chainlink faucet](https://faucets.chain.link/) to acquire testnet tokens.
1. Check the [Supported Networks page](/ccip/supported-networks) to confirm that USDC are supported for your lane. In this example, you will transfer tokens from _Avalanche Fuji_ to _Polygon Mumbai_ so check the list of supported tokens [here](/ccip/supported-networks/v1_2_0/testnet#avalanche-fuji-polygon-mumbai).
1. Use the [Circle faucet](https://faucet.circle.com/) to acquire USDC tokens on _Avalanche Fuji_.

> 1. Learn how to [fund your contract](/resources/fund-your-contract). This guide shows how to fund your contract in LINK, but you can use the same guide for funding your contract with any ERC20 tokens as long as they appear in the list of tokens in MetaMask.

Reviewer: ```suggestion
1. Learn how to [fund your contract](/resources/fund-your-contract). This guide shows how to fund your contract in LINK, but you can use the same guide for funding your contract with any ERC-20 tokens as long as they appear in the list of tokens in MetaMask.
```

        | \_token                    | <CopyText text="0x5425890298aed601595a70AB815c96711a31Bc65" code/><br /> The _USDC_ contract address at the source chain (_Avalanche Fuji_ in this example). You can find all the addresses for each supported blockchain on the [supported networks page](/ccip/supported-networks). |
        | \_amount                   | <CopyText text="1000000" code/> <br /> The token amount (_1 USDC_).                                                                                                                                                                                                                   |

    1.  Click on `transact` and confirm the transaction on MetaMask.
    1.  After the transaction is successful, record the transaction hash. Here is an [example](https://testnet.snowtrace.io/tx/0x6d54c98bb5b2892b78cbbc29967f4c0384d10dfc168755a4cf14489a02c11c65) of a transaction on _Avalanche Fuji_.

    <Aside type="note">
      During gas price spikes, your transaction might fail, requiring more than _0.5 LINK_ to proceed. If your

>       transaction fails, fund your contract with more _LINK_ tokens and try again.

Reviewer: Does the CCIP Explorer or any error message indicate when the transaction fails due to this reason? If so, mention it here

Author: @thedriftofwords unfortunately not
